File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(string& a, string& b) {
    if (a.size() == b.size()) return a > b;
    return a + b > b + a;
}

string solution(vector<int> numbers) {
    vector<string> vec;
    for (int num : numbers) vec.push_back(to_string(num));
    sort(vec.begin(), vec.end(), cmp);
    string res;
    for (auto v : vec) res.append(v);
    for (char c : res) if (c != '0') return res;
    return "0";
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public String solution(int[] numbers) {
        String res = String.join("", Arrays.stream(numbers)
            .mapToObj(Integer::toString)
            .sorted(new Comparator<String>(){
                @Override
                public int compare(String o1, String o2) {
                    return Integer.parseInt(o2 + o1)-Integer.parseInt(o1 + o2);
                }
            }).toArray(String[]::new));
        for (int i = 0; i < res.length(); i++) {
            if (res.charAt(i) != '0')
                return res;
        }
        return "0";
    }
}